server/server.js

```diff
@@ -27,3 +34,8 @@ app.listen(port, (error) => {
         console.info(`==> 🌎  Listening on port ${port}. Open up http://localhost:${port}/ in your browser.`);
     }
 });
+
+app.post('/api/newProblem', (req, res) => {
```

Comment:
At least for consistency in Express application - app.listen should be the last line in your server.js file
Order of middlewares matters.

And the most important thing here is that you need to remove this middleware, and use proper REST URL naming:
https://habrahabr.ru/post/38730/

so you already have 
app.use('/api/problems', problemsRouter);
Simply add POST handler to that router and voila.

router concept is added to make possible aka 'namespaces' for your REST API routing